File: packages/dart/lib/src/word/rand_nature.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Nature and its phenomena — sky, weather, water, land.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.nature] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randNature(language: WordLanguage.ko, count: 3); // [하늘, 노을, 바람]
/// randNature(language: WordLanguage.en, count: 3); // [Sky, Sunset, Breeze]
/// ```
List<String> randNature({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.nature,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);

File: packages/javascript/lib/word/randNature.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Nature and its phenomena — sky, weather, water, land.
 *
 * `randWord({ theme: 'nature' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randNature({ language: 'ko', count: 3 }); // ['하늘', '노을', '바람']
 * randNature({ language: 'en', count: 3 }); // ['Sky', 'Sunset', 'Breeze']
 */
export function randNature(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Nature words with the language and theme each one came from.
 *
 * @example
 * randNature({ language: 'ko', output: 'detail' });
 * // [{ word: '하늘', language: 'ko', theme: 'nature' }]
 */
export function randNature(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randNature(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('nature', options);
}
